File: 0800-letter-case-permutation/0800-letter-case-permutation.java
import java.util.*;
class Solution {
    public List<String> letterCasePermutation(String s) {
        List<String> out=new ArrayList<>();
        out.add("");
        for(char c : s.toCharArray()){
            List<String> tmp=new ArrayList<>();
            if(Character.isLetter(c)){
                for(String o:out){
                    tmp.add(o+Character.toLowerCase(c));
                    tmp.add(o+Character.toUpperCase(c));
                }
            }else{
                for(String o:out){
                    tmp.add(o+c);
                }
            }
            out=tmp;
        }
        return out;
    }
}

File: 0800-letter-case-permutation/0800-letter-case-permutation.py
class Solution(object):
    def letterCasePermutation(self, s):
        out=[""]
        for c in s:
            tmp=[]
            if c.isalpha():
                for o in out:
                    tmp.append(o+c.lower())
                    tmp.append(o+c.upper())
            else:
                for o in out:
                    tmp.append(o+c)
            out=tmp
        return out